pkg/plugin/state.go

```diff
@@ -678,6 +682,66 @@ func (e *AutoscaleEnforcer) handleVMDisabledScaling(logger *zap.Logger, podName
 	)
 }
 
+func (e *AutoscaleEnforcer) handlePodStartMigration(logger *zap.Logger, podName, migrationName util.NamespacedName, source bool) {
+	logger = logger.With(
+		zap.String("action", "VM pod start migration"),
+		zap.Object("pod", podName),
+		zap.Object("virtualmachinemigration", migrationName),
+	)
+
+	logger.Info("Handling VM pod migration start")
+
+	e.state.lock.Lock()
+	defer e.state.lock.Unlock()
+
+	pod, ok := e.state.podMap[podName]
+	if !ok {
+		logger.Warn("Cannot find pod in podMap")
+		return
+	}
+	logger = logger.With(zap.String("node", pod.node.name), zap.Object("virtualmachine", pod.vmName))
+
+	// Reset buffer to zero, remove from migration queue (if in it), and set pod's migrationState
+	cpuVerdict := collectResourceTransition(&pod.node.vCPU, &pod.vCPU).
+		handleStartMigration(source)
+	memVerdict := collectResourceTransition(&pod.node.memSlots, &pod.memSlots).
+		handleStartMigration(source)
+
+	pod.node.mq.removeIfPresent(pod)
+	pod.migrationState = &podMigrationState{name: migrationName}
+
+	logger.Info(
+		"Handled start of migration involving pod",
+		zap.Object("verdict", verdictSet{
+			cpu: cpuVerdict,
+			mem: memVerdict,
+		}),
+	)
+}
+
+func (e *AutoscaleEnforcer) handlePodEndMigration(logger *zap.Logger, podName, migrationName util.NamespacedName) {
+	logger = logger.With(
+		zap.String("action", "VM pod end migration"),
+		zap.Object("virtualmachinemigration", migrationName),
+	)
+
+	logger.Info("Handling VM pod migration end")
+
+	e.state.lock.Lock()
+	defer e.state.lock.Unlock()
+
+	pod, ok := e.state.podMap[podName]
+	if !ok {
+		logger.Warn("Cannot find pod in podMap")
+		return
+	}
+	logger = logger.With(zap.String("node", pod.node.name), zap.Object("virtualmachine", pod.vmName))
```

Comment:
I think it's a bit odd to make new logger objects rather than just adding some fields to a log message

Reply:
I'd say this is a case of localized weirdness in order to fit the broader pattern among similar functions — as such, going to keep it as is for now.